website/source/docs/agent/http/kv.html.markdown

```diff
@@ -142,3 +149,228 @@ DELETE request:
   synchronization primitives. Unlike `PUT`, the index must be greater than 0
   for Consul to take any action: a 0 index will not delete the key. If the index
   is non-zero, the key is only deleted if the index matches the `ModifyIndex` of that key.
+
+### <a name="txn"></a> /v1/txn
+
+Available in Consul 0.7 and later, this endpoint manages updates or fetches of
+multiple keys inside a single, atomic transaction. Only the `PUT` method is supported.
+
+By default, the datacenter of the agent receives the transaction; however, the dc
+can be provided using the "?dc=" query parameter. It is important to note that each
+datacenter has its own KV store, and there is no built-in replication between
+datacenters. If you are interested in replication between datacenters, look at the
+[Consul Replicate project](https://github.com/hashicorp/consul-replicate).
+
+The transaction endpoint supports the use of ACL tokens using the "?token=" query
+parameter.
+
+#### PUT Method
+
+The `PUT` method lets you submit a list of operations to apply to the key/value store
+inside a transaction. If any operation fails, the transaction will be rolled back and
+none of the changes will be applied.
+
+If the transaction doesn't contain any write operations then it will be fast-pathed
+internally to an endpoint that works like other reads, except that blocking queries
+are not currently supported. In this mode, you may supply the "?stale" or "?consistent"
+query parameters with the request to control consistency. To support bounding the
+acceptable staleness of data, read-only transaction responses provide the `X-Consul-LastContact`
+header containing the time in milliseconds that a server was last contacted by the leader node.
+The `X-Consul-KnownLeader` header also indicates if there is a known leader. These
+won't be present if the transaction contains any write operations, and any consistency
+query parameters will be ignored, since writes are always managed by the leader via
+the Raft consensus protocol.
+
+The body of the request should be a list of operations to perform inside the atomic
+transaction, which looks like this:
+
+```javascript
+[
+  {
+    "KV": {
+      "Verb": "<verb>",
+      "Key": "<key>",
+      "Value": "<Base64-encoded blob of data>",
+      "Flags": <flags>,
+      "Index": <index>,
+      "Session": "<session id>"
+    }
+  },
+  ...
+]
+```
+
+Up to 500 operations may be present in a single transaction.
```

Comment:
I wonder if we should reduce this to 64 or something initially for sanity to see how people use it. 500 seems like a pretty high limit.